ValveResourceFormat/Resource/ResourceTypes/ModelData/Attachments/Attachment.cs

```diff
@@ -0,0 +1,77 @@
+using System.Collections;
+using ValveResourceFormat.Serialization;
+using ValveResourceFormat.Serialization.KeyValues;
+
+namespace ValveResourceFormat.ResourceTypes.ModelData.Attachments
+{
+    public class Attachment : IEnumerable<Attachment.Influence>
+    {
+        public struct Influence
+        {
+            public string Name { get; init; }
+            public Vector3 Offset { get; init; }
+            public Quaternion Rotation { get; init; }
+            public double Weight { get; init; }
+        }
+
+        public string Name { get; init; }
+        public bool IgnoreRotation { get; init; }
+
+        private readonly Influence[] influences;
+        public Influence this[int i]
+        {
+            get
+            {
+                return influences[i];
+            }
+        }
```

Comment:
If @xPaw is ok with inline arrays. Source2 CAttachment has fields of fixed size 3. Kv3 also always stores 3 influences. 
```suggestion
        [System.Runtime.CompilerServices.InlineArray(3)]
        public struct Influence3
        {
            public Influence Influence;
        }

        public Influence3 Influences { get; init; }
```

That lets you index into attachment.Influences with bracket notation [] as well.

Reply:
This can't be indexed if it's a property, and if it's not a property it's throwing 'CA1051 Do not declare visible instance fields'